### PR TITLE
Aggregate-report fixes (part 1)

### PR DESCRIPTION
**Pull request by conniey**
Communication pom.xml
* Alphabetises communication modules.
  * Updates version metadata on jacoco to correct line

Root pom.xml
* Fixes broken pom.xml due to communication-administration not existing.
* Adds missing library entry for communication.

aggregate-report.yml
* Splits run into two jobs (Compliance and Generate).

[Edit]: May split into 3rd job for link verification.

Fixes https://github.com/Azure/azure-sdk-tools/issues/1423

### CODE REVIEW

File: eng/pipelines/aggregate-reports.yml [hunk 1]
```diff
@@ -128,3 +94,38 @@ jobs:
...
       displayName: 'Publish Report Artifacts'
       artifact: reports
+  - job: Compliance
+    timeoutInMinutes: 120
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: 'windows-2019'
+    steps:
+    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
```
Comment (alzimmermsft) [inline eng/pipelines/aggregate-reports.yml:104]: Component governance should remain in the other job as it requires the repository to be built and installed

Comment (conniey) [reply eng/pipelines/aggregate-reports.yml:104]: Does it? AFAIK, It scans the maven pom.xml files

File: eng/pipelines/aggregate-reports.yml [hunk 2]
```diff
@@ -128,3 +94,38 @@ jobs:
       condition: succeededOrFailed()
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:94]: Didn't you want to move the verify-links to the other job as well?

File: eng/pipelines/aggregate-reports.yml [hunk 3]
```diff
@@ -128,3 +94,38 @@ jobs:
...
       artifact: reports
+  - job: Compliance
+    timeoutInMinutes: 120
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: 'windows-2019'
+    steps:
+    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+      # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
```
Comment (sima-zhu) [inline eng/pipelines/aggregate-reports.yml:105]: Based on discussion, we can move the link check step here as it does not need repository to be built

File: eng/pipelines/aggregate-reports.yml [hunk 4]
```diff
@@ -128,3 +94,38 @@ jobs:
       condition: succeededOrFailed()
       displayName: 'Publish Report Artifacts'
       artifact: reports
+  - job: Compliance
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:97]: Maybe we should call this job "Analyze (no build)" and the other "Analyze (with build)" or something along those lines. Note from https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#job you can also add a displayName property to be able to use spaces and special chars in the name.

File: eng/pipelines/aggregate-reports.yml [hunk 5]
```diff
@@ -110,21 +80,54 @@ jobs:
...
+      condition: succeededOrFailed()
+      displayName: 'Publish Report Artifacts'
+      artifact: reports
+  - job: Compliance
+    displayName: Analyze (No build)
+    timeoutInMinutes: 120
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: 'ubuntu-18.04'
```
Comment (sima-zhu) [inline eng/pipelines/aggregate-reports.yml:97]: Cred scan is asking win os to run their tool. Better to double check whether cred scan is able to run under ubuntu

Comment (sima-zhu) [reply eng/pipelines/aggregate-reports.yml:97]: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=741047&view=logs&j=38d918fc-df5a-5870-2b80-6c17fdb13baf&t=1c955c51-9d5f-5f71-c490-e8dcdbd43fdf

It seems run into some errors. Can we switch back to win vm and test?

**Review by weshaggard — APPROVED** [review]
Changes look good to me assuming the aggregate-reports is passing. Thanks for taking the time to help refactor and fix these!

**Review by srnagar — APPROVED** [review]

**Review by sima-zhu — APPROVED** [review]
LGTM for the cred scan and link verification step.

**Pull request merged**